source/slang/slang-ir-specialize.cpp

```diff
@@ -373,7 +373,76 @@ struct SpecializationContext
         // represents a definition rather than a declaration.
         //
         if(!canSpecializeGeneric(genericVal))
+        {
```

Comment:
This will cause the specialization pass to specialize all custom derivative functions even when not using auto-diff.

Instead of preemptively specialize the custom func here, we should consider run specialize pass and auto-diff pass in a loop until no more changes are possible.